cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java

```diff
@@ -625,21 +607,20 @@ private static String writeDanglingLineBaseVoltage(DanglingLine danglingLine, St
         return baseVoltage.getId();
     }
 
-    private static String writeDanglingLineVoltageLevel(DanglingLine danglingLine, String substationId, String baseVoltageId, String cimNamespace, XMLStreamWriter writer) throws XMLStreamException {
-        // New VoltageLevel
-        String voltageLevelId = CgmesExportUtil.getUniqueId();
-        VoltageLevelEq.write(voltageLevelId, danglingLine.getNameOrId() + "_VL", Double.NaN, Double.NaN, substationId, baseVoltageId, cimNamespace, writer);
-
-        return voltageLevelId;
-    }
-
-    private static String writeDanglingLineConnectivity(DanglingLine danglingLine, String voltageLevelId, String cimNamespace, XMLStreamWriter writer,
+    private static String writeDanglingLineConnectivity(DanglingLine danglingLine, String cimNamespace, XMLStreamWriter writer,
                                                         CgmesExportContext context) throws XMLStreamException {
         String connectivityNodeId = null;
-        if (danglingLine.getTerminal().getVoltageLevel().getTopologyKind().equals(TopologyKind.NODE_BREAKER)) {
-            // New ConnectivityNode
-            connectivityNodeId = CgmesExportUtil.getUniqueId();
-            ConnectivityNodeEq.write(connectivityNodeId, danglingLine.getNameOrId() + "_NODE", voltageLevelId, cimNamespace, writer);
+        if (context.writeConnectivityNodes()) {
+            // We keep the connectivity node from the boundary definition as an alias in the dangling line
+            if (danglingLine.getAliasFromType(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.CONNECTIVITY_NODE).isPresent()) {
+                connectivityNodeId = context.getNamingStrategy().getCgmesIdFromAlias(danglingLine, Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.CONNECTIVITY_NODE);
+            } else {
+                // If no information about original boundary has been preserved in the IIDM model,
+                // we will create a new ConnectivityNode inside the voltage level of the dangling line network side
```

Comment:
As discussed, I would rather create a fictitious new voltage level.

Reply:
I discussed with @annetill and she agrees that we need a fictitious voltage level **and** a fictitious substation in this case. It is too weird otherwise...

Reply:
Fictitious containers (substation and voltage level) are created.